### ISSUE TITLE
the pps didn't count your last piece dropped

### ISSUE THREAD

**Issue by morningpee** [nullpomino/nullpomino#12]
```
(D) DragonOfFlames
11:03
dan did you see the videO?
(D) dhuang
11:03
yeah
(D) dhuang
11:03
it's just a calculation error i suppose
LapSiLap
11:03
Ye alex :s but i need to sleep :(
(D) DragonOfFlames
11:03
i guess so
(D) Alexandra
11:03
yeah it happens i think
(D) Alexandra
11:03
when i used to 100line cheese
(D) Alexandra
11:03
one line was odd
(D) Alexandra
11:04
so i tried calculating it and unless my math was wrong
(D) Alexandra
11:04
nullpo has an error
(D) Alexandra
11:04
but i think i have my cross multiplication down..
(D) Alexandra
11:04
;_; or not
(D) Alexandra
11:04
goodnight wendy
LapSiLap
11:05
Nite
(D) DragonOfFlames
11:06
are there any up to date records of sprint times at all? just wondering where i fall
(D) Alexandra
11:07
there's one by jkwon
(D) Alexandra
11:07
ill find it for you.. i guess..
(D) Alexandra
11:08
http://harddrop.com/forums/index.php?showtopic=6730&hl=jkwon
(D) DragonOfFlames
11:12
thanks!
(D) Alexandra
11:14
but the real question is why theres an error on nullpo
(D) Alexandra
11:14
isnt it a calculation?
Alexandra
11:38
there are only 2possibilities
Alexandra
11:39
either my math is really bad
Alexandra
11:39
and im wrong
Alexandra
11:40
or nullpo has error and my math isn't bad
Alexandra
11:41
but i think my math is bad
morningpee
12:30
@Alexandra if there is a bug in NullpoMino, please post it.
morningpee
12:30
https://github.com/nullpomino/nullpomino/issues
morningpee
12:30
nullpomino is actively developed
DragonOfFlames
12:39
@morning, we're not exactly sure but we're assuming theres a calculation error or maybe display issue somewhere
DragonOfFlames
12:40
i got a 24.81 40lines 100piece sprint but the PPS is 3.992
DragonOfFlames
12:40
it should be approx 4.03
DragonOfFlames
12:40
https://scontent.fewr1-1.fna.fbcdn.net/v/t34.0-12/14218313_1087474474664823_1549929587_n.png?oh=408e6a8456b6efa1828b24b34e53397a&oe=57CAE3AB
(D) TTO4 Winner
12:57
🤔
(D) pizzzapizza
1:13
😛
morningpee
1:25
@DragonOfFlames neat so take all of that info
morningpee
1:25
and put it here https://github.com/nullpomino/nullpomino/issues/new
morningpee
1:25
copy-paste is ok
(D) BentoBoxer
1:54
dragon I remember there being some old nullpo bug where the pps didn't count your last piece dropped
(D) BentoBoxer
1:55
when you divide 99 by 24.81 you get 3.99
(D) BentoBoxer
1:55
coincidence...probably
(D) Blink
9:08
homeeeee
(D) Blink
9:08
who wants to 1v1 practice in a bit
(D) Blink
9:08
where is the micro jkwon punman
(D) Blink
9:08
MicrojkwonPunMan
DragonOfFlames
9:24
oh i see, alright thanks bento
```

**Comment by jezevec10** [comment]
Displayed PPS and PPM values indeed don't count with the last placed block.

This is because when `LineRaceMode` (or any other mode) determines that the game has ended and calls `engine.ending = 1; engine.gameEnded();` the stats are no longer updated in `GameEngine.update()`.

Updating the statistics one last time in `engine.gameEnded()` should solve the problem.

Suggested fix is in #14

**Comment by morningpee** [comment]
Accepted pull request in a620e6befa. Closed.
